client/history/caching_redirector.go

```diff
@@ -57,25 +60,46 @@ type (
 		}
 
 		connections            connectionPool
+		goros                  goro.Group
 		historyServiceResolver membership.ServiceResolver
 		logger                 log.Logger
+		membershipUpdateCh     chan *membership.ChangedEvent
+		staleTTL               time.Duration
 	}
 )
 
+const cachingRedirectorListener = "cachingRedirectorListener"
+
 func newCachingRedirector(
 	connections connectionPool,
 	historyServiceResolver membership.ServiceResolver,
 	logger log.Logger,
+	staleTTL time.Duration,
 ) *cachingRedirector {
 	r := &cachingRedirector{
 		connections:            connections,
 		historyServiceResolver: historyServiceResolver,
 		logger:                 logger,
+		membershipUpdateCh:     make(chan *membership.ChangedEvent, 1),
+		staleTTL:               staleTTL,
 	}
 	r.mu.cache = make(map[int32]cacheEntry)
+
+	if r.staleTTL > 0 {
+		r.goros.Go(func(ctx context.Context) error {
+			r.eventLoop(ctx)
+			return nil
+		})
```

Comment:
make `r.eventLoop` return `error` so this can be just `r.goros.Go(r.eventLoop)`?